doc/source/roadmap-detailed.rst

```diff
@@ -0,0 +1,387 @@
+    In the future we can consider (soft-)deprecating the older API.
+
+The numerical integration functions are in good shape.  Support for integrating
+complex-valued functions and integrating multiple intervals (see `gh-3325
+<https://github.com/scipy/scipy/issues/3325>`__) could be added.
+
+
+interpolate
+```````````
+
+Ideas for new features:
+
+  - Spline fitting routines with better user control.
+  - Transparent tensor-product splines.
+  - NURBS support.
+  - Mesh refinement and coarsening of B-splines and corresponding tensor products.
+
+io
+``
+wavfile;
+
+    - PCM float will be supported, for anything else use ``audiolab`` or other
+      specialized libraries.
+    - Raise errors instead of warnings if data not understood.
+
+Other sub-modules (matlab, netcdf, idl, harwell-boeing, arff, matrix market)
+are in good shape.
+
+
+linalg
+``````
+``scipy.linalg`` is in good shape.  We have started requiring more recent
+LAPACK versions (minimum version increases from 3.1.0 to 3.4.0 in SciPy 1.2.0);
+we want to add support for newer features in LAPACK.
+
+Needed:
+
+  - Reduce duplication of functions with ``numpy.linalg``, make APIs consistent.
+  - ``get_lapack_funcs`` should always use ``flapack``
+  - Wrap more LAPACK functions
+  - One too many funcs for LU decomposition, remove one
+
+Ideas for new features:
+
+  - Add type-generic wrappers in the Cython BLAS and LAPACK
+  - Make many of the linear algebra routines into gufuncs
+
+
+misc
+````
+``scipy.misc`` will be removed as a public module.  Most functions in it have
+been moved to another submodule or deprecated.  The few that are left:
+
+  - ``doccer`` : move to ``scipy._lib`` (making it private)
+  - ``info``, ``who`` : these are NumPy functions
+  - ``derivative``, ``central_diff_weight`` : remove, possibly replacing them
+    with more extensive functionality for numerical differentiation.
+
+
+ndimage
+```````
+Underlying ``ndimage`` is a powerful interpolation engine.  Users come
+with an expectation of one of two models: a pixel model with ``(1,
+1)`` elements having centers ``(0.5, 0.5)``, or a data point model,
+where values are defined at points on a grid.  Over time, we've become
+convinced that the data point model is better defined and easier to
+implement, but this should be clearly communicated in the documentation.
+
+More importantly, still, SciPy implements one *variant* of this data
+point model, where datapoints at any two extremes of an axis share a
+spatial location under *periodic wrapping* mode.  E.g., in a 1D array,
+you would have ``x[0]`` and ``x[-1]`` co-located.  A very common
+use-case, however, is for signals to be periodic, with equal spacing
+between the first and last element along an axis (instead of zero
+spacing).  Adding wrapping modes for this use-case and disambiguating
+them from the existing implementation should address several issues,
+including #1323, #1903, #2045 and #2640.
```

Comment:
The basis for this was done in #8537, but the interpolation routines need updating. I think that should not be too difficult. @jaimefrio Comment?

Reply:
The ndimage interpolation might be considered a mix of the two conventions, which it used depends on the spline order. That is, for odd orders the knot points are at the same location as the data points, while for even orders the knot points are midway between the data points. That is convenient, in that the number of needed boundary conditions is always even and can be rationally split between the beginning and end of the data.

I think the distinction between data point models is artificial. For images the data point is simply taken at the center of the pixel, which makes sense as the output is the sum over the pixel. Even in other applications, understanding the result needs some familiarity with the bandwidth of the signal vs the measurement device and sample frequency.

Reply:
The required new modes were indeed added recently, thanks @charris. 

> I think the distinction between data point models is artificial. 

Not sure I agree with this. Reality is more complex, but it does help to think about these two modes. IIRC we had a mix of both modes even for the default of odd order.

Reply:
I should probably clarify. We have five types of boundary conditions, the spline coefficients can now be computed for all of them. I believe that interpolation also works for all of them (@jaimefrio ). The "mirror" and "reflect" modes correspond to the boundary conditions for DCT-I and DCT-II. The different boundary conditions all use the same data points and all can be evaluated outside the interval spanned by the data points, but their behavior outside that interval varies depending on the mode. In "reflect" mode, which is generally preferred for numerical reasons among others, the interval [-.5, 0] has content that may be of interest and cannot be obtained by simply mirroring into interior points. I don't regard this primarily a  difference between pixel/data views, but rather as different boundary conditions. The pixel geometry might lead one to prefer the 'reflect' mode, but really, at the edge, the interpolated results are going to be suspect in any case.

Due to the way the splines are implemented, the correspondence between data points and knot points varies as pointed out previously. That may be counter intuitive, but the major effect is that differentiating or integrating the splines does not yield the knot points expected by the current interpolation routines. I don't know if there is a plan to fix that.